torch/csrc/jit/tensorexpr/llvm_jit.cpp

```diff
@@ -171,7 +171,16 @@ class TORCH_API PytorchLLVMJITImpl {
   }
 
   JITSymbol findSymbol(const std::string Name) {
+#if LLVM_VERSION_MAJOR >= 15
+    // Starting with llvm-15, LLJIT::lookup returns an address rather than a
+    // symbol. Even though an address is what we ultimately we want, we also
+    // want to avoid churning our internal APIs, so we wrap the returned address
+    // in a fake JITSymbol.
+    auto result = assertSuccess(LLJ->lookup(Name));
+    return JITSymbol(result.getValue(), JITSymbolFlags());
```

Comment:
@bertmaher Was this code tested with LLVM15? We see the following error when trying to compile this code with LLVM15:
```
[2022-08-04T05:14:13.322Z] [ 82%] Building CXX object caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/tensorexpr/mem_dependency_checker.cpp.o
[2022-08-04T05:14:13.322Z] [91m/var/lib/jenkins/pytorch/torch/csrc/jit/tensorexpr/llvm_jit.cpp: In member function âllvm::JITSymbol llvm::orc::PytorchLLVMJITImpl::findSymbol(std::string)â:
[2022-08-04T05:14:13.322Z] [0m[91m/var/lib/jenkins/pytorch/torch/csrc/jit/tensorexpr/llvm_jit.cpp:180:29: error: âclass llvm::JITEvaluatedSymbolâ has no member named âgetValueâ
[2022-08-04T05:14:13.322Z] [0m[91m  180 |     return JITSymbol(result.getValue(), JITSymbolFlags());
[2022-08-04T05:14:13.322Z] [0m[91m      |                             ^~~~~~~~
[2022-08-04T05:14:13.322Z] [0m[91mmake[2]: *** [caffe2/CMakeFiles/torch_cpu.dir/build.make:10057: caffe2/CMakeFiles/torch_cpu.dir/__/torch/csrc/jit/tensorexpr/llvm_jit.cpp.o] Error 1
[2022-08-04T05:14:13.322Z] make[2]: *** Waiting for unfinished jobs....
[2022-08-04T05:14:31.545Z] [0m[91mmake[1]: *** [CMakeFiles/Makefile2:5799: caffe2/CMakeFiles/torch_cpu.dir/all] Error 2
[2022-08-04T05:14:31.545Z] [0m[91mmake: *** [Makefile:146: all] Error 2
```

Reply:
https://llvm.org/doxygen/classllvm_1_1JITEvaluatedSymbol.html

Reply:
This is weird, why is it `JITEvaluatedSymbol`? From the docs `lookup` returns `ExecutorAddr`:
https://llvm.org/doxygen/classllvm_1_1orc_1_1LLJIT.html#a12731a7da7f90e0596518987df3cad65
And it has `getValue` method, which we're using here:
https://llvm.org/doxygen/classllvm_1_1orc_1_1ExecutorAddr.html#a4ee908fb9052f020e3c50e3f1a7d81c5

Reply:
@ZolotukhinM My team is looking into that. Meanwhile, can you please share whether this PR was tested with llvm-trunk or LLVM15 in CI?

Reply:
Sorry, I dont know that (but I'd suppose yes).

Reply:
I think we're good here. Our internal "llvm15" compiler was based off of an early llvm15-based build aka one which did not include the llvm patch that necessitated this PR. We are re-testing with an updated llvm15-based compiler.